AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Migration/MutationSyncMetadataMigrationDelegate+SQLite.swift

```diff
@@ -103,8 +103,8 @@ final class SQLiteMutationSyncMetadataMigrationDelegate: MutationSyncMetadataMig
             sql += "SELECT id, \'\(modelName)\' as tableName FROM \(modelName)"
         }
         sql = "INSERT INTO \(MutationSyncMetadataMigration.MutationSyncMetadataCopy.modelName) (id,deleted,lastChangedAt,version) " +
-        "select models.tableName || '|' || mm.id, mm.deleted, mm.lastChangedAt, mm.version " +
-        "from MutationSyncMetadata mm INNER JOIN (" + sql + ") as models on mm.id=models.id"
+        "select mm.id, mm.deleted, mm.lastChangedAt, mm.version " +
+        "from MutationSyncMetadata mm INNER JOIN (" + sql + ") as models on mm.id=models.tableName || '|' || models.id"
```

Comment:
what backfill does is if the existing data on the app is not in the format of "<modelName>|<modelId>" then backfill it to this format. I don't think we can change this logic, if we change it it's like we might as well remove the migration script all together. doing so will be a one-way door because some existing apps will break if the developer upgrades to 2.0.0 and ships the app.